src/main/scala/org/allenai/plugins/ReleasePlugin.scala

```diff
@@ -14,32 +14,51 @@ object ReleasePlugin extends AutoPlugin {
 
   override def requires: Plugins = plugins.JvmPlugin
 
-  private def todayVersion: String = {
-    val df = new SimpleDateFormat("yyyy.MM.dd")
-    df.format(new Date())
-  }
+  object DateVersion {
+    private def todayVersion: String = {
+      val df = new SimpleDateFormat("yyyy.MM.dd")
+      df.format(new Date())
+    }
 
-  val VersionPattern = """(\d+\.\d+\.\d+)-(\d+)(?:-SNAPSHOT)?""".r
+    val VersionPattern = """(\d+\.\d+\.\d+)-(\d+)(?:-SNAPSHOT)?""".r
 
-  def incrementVersion(prev: String): String = {
-    prev match {
-      case VersionPattern(prefix, num) => s"${prefix}-${num.toInt + 1}"
-      case _ => throw new IllegalStateException(s"Invalid version number: ${prev}")
+    def incrementVersion(prev: String): String = {
+      prev match {
+        case VersionPattern(prefix, num) => s"${prefix}-${num.toInt + 1}"
+        case _ => throw new IllegalStateException(s"Invalid version number: ${prev}")
+      }
+    }
+
+    def releaseVersion(version: String) = {
+      val today = todayVersion
+      if (version.startsWith(today)) {
+        version.replace("-SNAPSHOT", "")
+      } else {
+        s"${today}-0"
+      }
+    }
+
+    def nextVersion(version: String) = {
+      s"${incrementVersion(version)}-SNAPSHOT"
+    }
+  }
+
+  object SemanticVersion {
+    def nextVersion(version: String) = {
+      sbtrelease.Version(version).
+        map(_.bump(Version.Bump.Next).asSnapshot.string).
+        getOrElse(sbtrelease.versionFormatError)
+    }
+    def releaseVersion(version: String) = {
+      sbtrelease.Version(version).
+        map(_.withoutQualifier.string).
+        getOrElse(sbtrelease.versionFormatError)
```

Comment:
Instead of objects, I think it would be better to provide ready-to-use `Seq[Def.Setting[_]]` here. Otherwise you force the build author to know to override two settings. Something like:

```scala
object autoImport {
  object Release {
    val dateVersioning: Seq[Def.Setting[_]] = Seq(
      releaseVersion := { version => DateVersion.releaseVersion(version) },
      nextVersion := { version => DateVersion.nextVersion(version) }
    )

    val semanticVersioning: Seq[Def.Setting[_]] = Seq(
      releaseVersion := { version => SemanticVersion.releaseVersion(version) },
      nextVersion := { version => SemanticVersion.nextVersion(version) }
    )
  }
}

override def projectSettings: Seq[Def.Setting[_]] = autoImport.Release.dateVersioning
```

Then in `build.sbt` you'd just do this:

```scala
enablePlugins(LibraryPlugin)
Release.semanticVersioning
```